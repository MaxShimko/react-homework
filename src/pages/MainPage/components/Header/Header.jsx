import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';

import './style.scss';
import Search from "../../../../components/Search/Search.jsx";

export default function Header({onSearch}) {
    return (
        <div className="main-page-header">
            <Container>
                <Row>
                    <span className="logotype">
                        <span className="bold">netflix</span>roulette
                    </span>
                </Row>
                <div className="delimiter-6" />
                <Row>
                    <Col md={{span: 10, offset: 1}}>
                        <h1>Find your movie</h1>
                        <div className="delimiter-2" />
                        <Search id="test_header_search" onSearch={onSearch.bind(this)} />
                    </Col>
                </Row>
                <div className="delimiter-10" />
            </Container>
        </div>
    );
};
