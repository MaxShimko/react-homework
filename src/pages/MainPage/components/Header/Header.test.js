import React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';
import {describe, expect, it} from '@jest/globals';
import toJson from 'enzyme-to-json';

describe('<Header />', () => {
    it('renders without crashing', () => {
        const onSearch = jest.fn();
        const component = shallow(<Header onSearch={onSearch}/>);
        expect(toJson(component)).toMatchSnapshot();
    });
});
