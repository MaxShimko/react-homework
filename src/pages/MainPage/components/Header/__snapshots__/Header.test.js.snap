// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders without crashing 1`] = `
<div
  className="main-page-header"
>
  <Container
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <span
        className="logotype"
      >
        <span
          className="bold"
        >
          netflix
        </span>
        roulette
      </span>
    </Row>
    <div
      className="delimiter-6"
    />
    <Row
      noGutters={false}
    >
      <Col
        md={
          Object {
            "offset": 1,
            "span": 10,
          }
        }
      >
        <h1>
          Find your movie
        </h1>
        <div
          className="delimiter-2"
        />
        <Search
          id="test_header_search"
          onSearch={[Function]}
        />
      </Col>
    </Row>
    <div
      className="delimiter-10"
    />
  </Container>
</div>
`;
